### PR TITLE
feat: add composables for migration and update portal settings

### PR DESCRIPTION
**Pull request by scmmishra**
This PR updates portal settings to use the new composition API along with the `<script setup>` syntax. This PR is supposed to be a reference on how to migrate existing views to Vue 3 compatible syntax without breaking any functionality

## Core changes

- [x] Add composable for `store`
- [x] Add composable for `route`
- [x] Add composable for `showAlert`
- [x] Add composable for `$track`

## Portal Changes

- [x] Fix layout spacing in the settings page
- [x] Set max width to `5xl` for tables and `2xl` for forms
- [x] Use `<script setup>` syntax wherever possible
- [x] Move styles to tailwind wherever possible
- [ ] ~Move `portalMixin` to a composable (Not sure if this is possible :P)~ Can do in a separate PR

> The `useI18n` was already added in a separate PR [here](https://github.com/chatwoot/chatwoot/pull/9278/files#diff-6606006327b16be1b7fcc59a2de67c298ec44b4dc8859d0915284b4afb26a99b)

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+  name: 'PortalCustomization',
+});
+
+const getters = useStoreGetters();
+const route = useRoute();
+const router = useRouter();
+const store = useStore();
+const { t } = useI18n();
+
+const uiFlags = getters['portals/uiFlagsIn'];
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue:21]: Should we make use of computed to ensure that `uiFlags` is reactive and updates correctly as per the Vuex store's state changes.
```suggestion
const uiFlags = computed(() => getters['portals/uiFlagsIn']);
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+  const portalSlug = route.params.portalSlug;
+  let alertMessage = '';
+  try {
+    await store.dispatch('portals/update', {
+      portalSlug,
+      ...portalObj,
+    });
+    alertMessage = t('HELP_CENTER.PORTAL.ADD.API.SUCCESS_MESSAGE_FOR_UPDATE');
+
+    useTrack(PORTALS_EVENTS.ONBOARD_CUSTOMIZATION, {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue:44]: Getting this error after submitting the customization form. I think something wrong with `useTrack` method.
CleanShot 2024-04-29 at 10 20 12@2x

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsCustomizationForm.vue [hunk 3]
```diff
@@ -1,68 +1,146 @@
...
+  emit('submit', portal);
+}
+
+onMounted(() => {
+  updateDataFromStore();
+});
+</script>
+
 <template>
-  <div
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsCustomizationForm.vue:2]: Please fix this style issues.
CleanShot 2024-04-29 at 10 22 59@2x

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue:21]: The value is already wrapped a `computed` in the composable

```js
export const useStoreGetters = () => {
  const store = useStore();
  return Object.fromEntries(
    Object.keys(store.getters).map(getter => [
      getter,
      computed(() => store.getters[getter]),
    ])
  );
};
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue (hunk 2 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/helpcenter/pages/portals/PortalCustomization.vue]: Good catch, fixed it

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsCustomizationForm.vue (hunk 3 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsCustomizationForm.vue:2]: Fixed this

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**